actionview/lib/action_view/dependency_tracker.rb

```diff
@@ -142,8 +145,20 @@ def add_static_dependency(dependencies, dependency)
           end
         end
 
+        def resolve_directories(wildcard_dependencies)
+          wildcard_dependencies.map do |query|
+            DependencyTracker.view_paths.find_all_with_query(query).map do |template|
+              "#{File.dirname(query)}/#{File.basename(template).split('.').first}"
+            end
+          end
+        end
+
         def explicit_dependencies
-          source.scan(EXPLICIT_DEPENDENCY).flatten.uniq
+          dependencies = source.scan(EXPLICIT_DEPENDENCY).flatten.uniq
+
+          wildcards, explicits = dependencies.partition { |dependency| dependency[-1] == '*' }
+
+          explicits + resolve_directories(wildcards).flatten.uniq
```

Comment:
What are you trying to call `uniq`? The merge of `explicits` with `resolve_directories(wildcards)` or just the `resolve_directories(wildcards)`? Because I think you are calling on the latter.